Kodi fails to build with Linux 5.1-rc1

## Bug report
### Describe the bug

When building Kodi against Linux 5.1-rc1, Kodi fails to build as follows:
```
In file included from
/home/neil/projects/scratch/alternates/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.1-devel-next/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/linux/kernel.h:5,
                 from
/home/neil/projects/scratch/alternates/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.1-devel-next/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/sys/sysinfo.h:24,
                 from ../xbmc/platform/linux/PlatformDefs.h:44,
                 from ../xbmc/XBDateTime.h:12,
                 from ../xbmc/utils/StringUtils.h:35,
                 from ../xbmc/utils/log.h:15,
                 from ../xbmc/filesystem/DllLibCurl.cpp:13:
/home/neil/projects/scratch/alternates/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.1-devel-next/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/linux/sysinfo.h:9:2:
error: '__kernel_long_t' does not name a type; did you mean
'__syscall_ulong_t'?
  __kernel_long_t uptime;  /* Seconds since boot */
  ^~~~~~~~~~~~~~~
  __syscall_ulong_t
/home/neil/projects/scratch/alternates/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.1-devel-next/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/linux/sysinfo.h:10:2:
error: '__kernel_ulong_t' does not name a type; did you mean
'__syscall_ulong_t'?
  __kernel_ulong_t loads[3]; /* 1, 5, and 15 minute load averages */
  ^~~~~~~~~~~~~~~~
  __syscall_ulong_t
```

The change in behaviour is due to this 5.1-rc1 commit: https://github.com/torvalds/linux/commit/a623a7a1a5670c25a16881f5078072d272d96b71

However this change shouldn't be an issue for user space apps, but for Kodi it is.

## Expected Behavior

Kodi builds successfully with Linux 5.1-rc1 (it builds fine with Linux 5.0.2).


## Actual Behavior

Kodi fails due to undefined type.


## Possible Fix

I pinged this issue over to the [glibc mailing list](https://sourceware.org/ml/libc-help/2019-03/msg00010.html) and they concluded this is a [Kodi issue](https://sourceware.org/ml/libc-help/2019-03/msg00014.html) that Kodi needs to fix as it misuses headers.



### To Reproduce
Steps to reproduce the behavior:
1. Build LibreELEC with kernel 5.1-rc1


### Debuglog

Kodi build log: http://ix.io/1DXy


### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [X] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Ubuntu 17.10
 - Kodi version: Kodi 18 master (76db73d1020c91378572fc4fb2b627b71293ea0b)

<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

Not that now fails for a different reason: http://ix.io/1DYA
```
../xbmc/filesystem/CurlFile.cpp: In member function 'virtual bool XFILE::CCurlFile::Exists(const CURL&)':
../xbmc/filesystem/CurlFile.cpp:1254:9: error: reference to 'curl_slist' is ambiguous
         curl_slist *list = NULL;
         ^~~~~~~~~~
In file included from ../xbmc/filesystem/CurlFile.cpp:9:
../xbmc/filesystem/CurlFile.h:21:10: note: candidates are: 'struct XCURL::curl_slist'
   struct curl_slist;
          ^~~~~~~~~~
In file included from ../xbmc/filesystem/DllLibCurl.h:21,
                 from ../xbmc/filesystem/CurlFile.cpp:33:
/home/neil/projects/scratch/alternates/LibreELEC.tv/build.LibreELEC-Generic.x86_64-9.1-devel-next/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/curl/curl.h:2413:8: note:                 'struct curl_slist'
 struct curl_slist {
        ^~~~~~~~~~
```

This is how `xbmc/filesystem/DllLibCurl.h` looks now: http://ix.io/1DYC

--- a/xbmc/filesystem/CurlFile.cpp
+++ b/xbmc/filesystem/CurlFile.cpp
@@ -1818,7 +1818,7 @@ std::string CCurlFile::GetRedirectURL()
 std::string CCurlFile::GetInfoString(int infoType)
 {
   char* info{};
-  CURLcode result = g_curlInterface.easy_getinfo(m_state->m_easyHandle, static_cast<XCURL::CURLINFO> (infoType), &info);
+  CURLcode result = g_curlInterface.easy_getinfo(m_state->m_easyHandle, static_cast<CURLINFO> (infoType), &info);
   if (result != CURLE_OK)
   {
     CLog::Log(LOGERROR, "Info string request for type {} failed with result code {}", infoType, result);
@@ -1894,9 +1894,9 @@ bool CCurlFile::GetContentType(const CURL &url, std::string &content, const std:
 bool CCurlFile::GetCookies(const CURL &url, std::string &cookies)
 {
   std::string cookiesStr;
-  struct curl_slist*     curlCookies;
-  XCURL::CURL_HANDLE*    easyHandle;
-  XCURL::CURLM*          multiHandle;
+  curl_slist* curlCookies;
+  CURL_HANDLE* easyHandle;
+  CURLM* multiHandle;
 
   // get the cookies list
   g_curlInterface.easy_acquire(url.GetProtocol().c_str(),
@@ -1905,7 +1905,7 @@ bool CCurlFile::GetCookies(const CURL &url, std::string &cookies)
   if (CURLE_OK == g_curlInterface.easy_getinfo(easyHandle, CURLINFO_COOKIELIST, &curlCookies))
   {
     // iterate over each cookie and format it into an RFC 2109 formatted Set-Cookie string
-    struct curl_slist* curlCookieIter = curlCookies;
+    curl_slist* curlCookieIter = curlCookies;
     while(curlCookieIter)
     {
       // tokenize the CURL cookie string

--- a/xbmc/filesystem/CurlFile.h
+++ b/xbmc/filesystem/CurlFile.h
@@ -14,12 +14,9 @@
 #include <string>
 #include "utils/HttpHeader.h"
 
-namespace XCURL
-{
-  typedef void CURL_HANDLE;
-  typedef void CURLM;
-  struct curl_slist;
-}
+typedef void CURL_HANDLE;
+typedef void CURLM;
+struct curl_slist;
 
 namespace XFILE
 {
@@ -97,8 +94,8 @@ namespace XFILE
       public:
           CReadState();
           ~CReadState();
-          XCURL::CURL_HANDLE* m_easyHandle;
-          XCURL::CURLM* m_multiHandle;
+          CURL_HANDLE* m_easyHandle;
+          CURLM* m_multiHandle;
 
           CRingBuffer m_buffer; // our ringhold buffer
           unsigned int m_bufferSize;
@@ -121,8 +118,8 @@ namespace XFILE
           CHttpHeader m_httpheader;
           bool IsHeaderDone(void) { return m_httpheader.IsHeaderDone(); }
 
-          struct XCURL::curl_slist* m_curlHeaderList;
-          struct XCURL::curl_slist* m_curlAliasList;
+          curl_slist* m_curlHeaderList;
+          curl_slist* m_curlAliasList;
 
           size_t ReadCallback(char *buffer, size_t size, size_t nitems);
           size_t WriteCallback(char *buffer, size_t size, size_t nitems);

--- a/xbmc/filesystem/DllLibCurl.h
+++ b/xbmc/filesystem/DllLibCurl.h
@@ -17,12 +17,12 @@
 #include <type_traits>
 #include <vector>
 
-/* put types of curl in namespace to avoid namespace pollution */
-namespace XCURL
-{
 #define CURL CURL_HANDLE
 #include <curl/curl.h>
 #undef CURL
+
+namespace XCURL
+{
 
 class DllLibCurl
 {
@@ -57,8 +57,8 @@ public:
   CURLMcode multi_timeout(CURLM* multi_handle, long* timeout);
   CURLMsg* multi_info_read(CURLM* multi_handle, int* msgs_in_queue);
   CURLMcode multi_cleanup(CURLM* handle);
-  struct curl_slist* slist_append(struct curl_slist* list, const char* to_append);
-  void slist_free_all(struct curl_slist* list);
+  curl_slist* slist_append(curl_slist* list, const char* to_append);
+  void slist_free_all(curl_slist* list);
   const char* easy_strerror(CURLcode code);
 };
 

